### DISCUSSION TITLE
Closure transaction's TransactionError and it's wrapped error

### DISCUSSION THREAD

**Discussion by szattila98** [SeaQL/sea-orm#859] [Q&A] [answered]
Hello, I have been trying to use the closure transaction API, mentioned in the [docs](https://www.sea-ql.org/SeaORM/docs/advanced-query/transaction/) as the preferred option, and on error the closure returns a `TransactionError<WrappedError>`. `WrappedError` is the type I specify in the `E` spot within the turbofish `::<_, T, E>` if I understand it right. My question is, how can I unwrap the `WrappedError` from the `TransactionError`? Also what is the first type in the turbofish, the `_` ?

**Comment by szattila98** [comment, marked as answer]
Oh I missed the part that the TransactionError is an enum, so it is easy to unwrap with match. The _ in the tutorial turbofish is probably the transaction closure signature if I understand it correctly.

**Comment by billy1624** [comment]
Thanks for the updates! @szattila98 :D
